scripts/release/stage-release.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+set -ex
+
+# Stages a release by putting everything that should be packaged and released
+# into the ./deploy folder. This script should be run from the root of the
+# material2 repo.
+
+# Make sure you are not running `ng serve` or `ng build --watch` when running this.
+
+
+# Clear dist/ and deploy/ so that we guarantee there are no stale artifacts.
+rm -rf ./dist
+rm -rf ./deploy
+
+# For packaging for npm only, replace the TypeScript module format to commonjs
+TS_CONFIG_ORIGINAL=$(cat ./src/tsconfig.json)
+SYSTEM_FORMAT="\"module\": \"system\","
+COMMONJS_FORMAT="\"module\": \"commonjs\","
+TS_CONFIG_MODIFIED=${TS_CONFIG_ORIGINAL/${SYSTEM_FORMAT}/${COMMONJS_FORMAT}}
+echo "$TS_CONFIG_MODIFIED" > ./src/tsconfig.json
+
+# Perform a build with the modified tsconfig.json.
+ng build
+
+# Return tsconfig.json to its original state.
+echo "$TS_CONFIG_ORIGINAL" > ./src/tsconfig.json
+
+# Inline the css and html into the component ts files.
+./node_modules/gulp/bin/gulp.js inline-resources
+
+# deploy/ serves as a working directory to stage the release.
+mkdir deploy
+
+# Copy all components/ to deploy/ and replace "../core" with just "core" since each
+# component directory will now live as a sibling to core/
+cp -R ./dist/components/* ./deploy/
+find ./deploy -type f \( -name "*.js" -o -name "*.ts" \) -exec sed -i '' 's|\.\./core|core|g' {} \;
```

Comment:
Nope, this is actually changing the imports. I'm doing this so that we can keep our source code organized into `components/` and `core/`, but actually release with the components as siblings to core. Since the components include core via relative path, we have "un up" the path to core by one. 

I know it seems weird, but it will be safe if we stick to our current directory structure.